pkg/maps/egressmap/policy.go

```diff
@@ -33,41 +39,74 @@ type EgressPolicyVal4 struct {
 	GatewayIP types.IPv4 `align:"gateway_ip"`
 }
 
-// egressPolicyMap is the internal representation of an egress policy map.
-type egressPolicyMap struct {
-	*ebpf.Map
+type PolicyConfig struct {
+	// EgressGatewayPolicyMapMax is the maximum number of entries
+	// allowed in the BPF egress gateway policy map.
+	EgressGatewayPolicyMapMax int
 }
 
-// initEgressPolicyMap initializes the egress policy map.
-func initEgressPolicyMap(policyMapName string, maxPolicyEntries int, create bool) error {
-	var m *ebpf.Map
+var DefaultPolicyConfig = PolicyConfig{
+	EgressGatewayPolicyMapMax: 1 << 14,
+}
 
-	if create {
-		m = ebpf.NewMap(&ebpf.MapSpec{
-			Name:       policyMapName,
-			Type:       ebpf.LPMTrie,
-			KeySize:    uint32(unsafe.Sizeof(EgressPolicyKey4{})),
-			ValueSize:  uint32(unsafe.Sizeof(EgressPolicyVal4{})),
-			MaxEntries: uint32(maxPolicyEntries),
-			Pinning:    ebpf.PinByName,
-		})
+func (def PolicyConfig) Flags(flags *pflag.FlagSet) {
+	flags.Int("egress-gateway-policy-map-max", def.EgressGatewayPolicyMapMax, "Maximum number of entries in egress gateway policy map")
+}
 
-		if err := m.OpenOrCreate(); err != nil {
-			return err
-		}
-	} else {
-		var err error
+// PolicyMap is used to communicate EGW policies to the datapath.
+type PolicyMap interface {
+	Lookup(sourceIP net.IP, destCIDR net.IPNet) (*EgressPolicyVal4, error)
+	Update(sourceIP net.IP, destCIDR net.IPNet, egressIP, gatewayIP net.IP) error
+	Delete(sourceIP net.IP, destCIDR net.IPNet) error
+	IterateWithCallback(EgressPolicyIterateCallback) error
+}
+
+// policyMap is the internal representation of an egress policy map.
+type policyMap struct {
+	m *ebpf.Map
+}
 
-		if m, err = ebpf.LoadRegisterMap(policyMapName); err != nil {
-			return err
-		}
+func createPolicyMapFromDaemonConfig(daemonConfig *option.DaemonConfig, lc hive.Lifecycle, cfg PolicyConfig) bpf.MapOut[PolicyMap] {
+	if !daemonConfig.EnableIPv4EgressGateway {
```

Comment:
Maybe we can get away without this? Ties into the whole "optional hive feature".